src/main/java/com/faforever/api/map/MapService.java

```diff
@@ -97,7 +97,7 @@ public MapNameValidationResponse requestMapNameValidation(String mapName) {
         .mapToInt(MapVersion::getVersion)
         .max()
         .orElse(1))
-      .orElse(1);
+      .orElse(2);
```

Comment:
the +1 must be applied on the max value `.map(i -> i+1)`

The .orElse part covers the case, that there is no existing map Version, in that case 1 is the next version